spec/pdfs/siren_infos_pdf_spec.rb

```diff
@@ -148,4 +148,28 @@
       its(:size) { is_expected.to eq 15 }
     end
   end
+
+  describe 'production issue' do
+    let(:dossier) { DossierEntreprise.new.tap(&:save) }
+
+    before do
+      create :representant, dossier_entreprise: dossier, type_representant: 'P. Physique', nom_patronyme: 'BUGGY', prenoms: 'spacing'
+    end
+
+    it { is_expected.to include('Nom prénoms: BUGGY spacing') }
+  end
+
+  context 'wrong expected values' do
+    subject(:pdf) { described_class.new dossier }
+    let(:dossier) { DossierEntreprise.new.tap(&:save) }
+
+    before do
+      create :representant, dossier_entreprise: dossier, type_representant: 'P. Banale'
+    end
+
+    it 'raises an error' do
+      expect(Rails.logger).to receive(:error).with('Unhandled type_representant')
+      pdf.render
+    end
+  end
```

Comment:
Pareil, ici il y a 12 lignes de codes pour un example group contenant un `subject` et un hook `before` pour un unique test. A la base, ces helpers servent à supprimer les répétitions dans les bloc `it`, et non pas comme manière obligatoire de setup les tests. 

Donc comme pour plus haut je partirai sur un bloc `it` unique, avec une description un peu verbeuse certes mais qui permettrait que le comportement testé saute aux yeux : 

```ruby 
it 'raises an error when :type_representant is unexpected' do 
  ...
end 
```

Pour moi, le fait que tu sois obligé de faire une écriture en base pour tester la génération du PDF c'est un code smell, mais c'est une histoire pour une autre PR, je vais créer une issue pour qu'on en discute.

Reply:
1000 fois mieux ❤️